posydon/binary_evol/binarystar.py

```diff
@@ -286,10 +286,12 @@ def reset(self, properties=None):
 
     def update_star_states(self):
         """Update the states of the two stars in the binary."""
-        self.star_1.state = check_state_of_star(
-            self.star_1, star_CO=self.star_1.state in ["WD", "BH", "NS"])
-        self.star_2.state = check_state_of_star(
-            self.star_2, star_CO=self.star_2.state in ["WD", "BH", "NS"])
+        if self.star_1.state != 'massless_remnant':
+            self.star_1.state = check_state_of_star(
+                self.star_1, star_CO=self.star_1.state in ["WD", "BH", "NS"])
```

Comment:
Just a question: is there a reason for the order here? Otherwise I'd suggest to better always use `["WD", "NS", "BH"]` (same at other instances, too), which gives a mass/compactness order to the CO_types.

Reply:
This was the order the code was in originally. I only added the if statements. happy to change the order, while we're at it.